src/schemaGenerator.ts

```diff
@@ -281,6 +281,14 @@ function addResolveFunctionsToSchema(schema: GraphQLSchema, resolveFunctions: IR
         return;
       }
 
+      if (typeName === 'JSON') {
```

Comment:
This is wrong as well,
you are not adding the ability to add any custom scalar type, but just the one named JSON.
the patch should just be able to merge any already existing GraphQLObject (Or just Scalar? - @helfer what do you think?) with the schema.